src/Components/firebase-access.jsx

```diff
@@ -3,465 +3,386 @@ import "firebase/auth";
+        let Industries = await db.collection("Ticker-Info").doc("Stock").collection("Stocks")
+            .where("Industry","==",Industry)
+            .where("IndustryPos","<=", cutoff)
+            .orderBy("IndustryPos").limit(NumOfSymbols).get()
+        Industries.forEach(function(doc){
+            symbols.push(doc.data().Symbol)
+        })
+
+    }
+    else if(Sector !== null){
+
+        let SectorInfo =  await db.collection("Ticker-Info").doc("Sector").get();
+        let numOfSectors= SectorInfo.data().Sector[Sector];
+        let cutoff = Math.floor((Math.random()  * (numOfSectors - NumOfSymbols))+NumOfSymbols);
+        let Sectors = await db.collection("Ticker-Info").doc("Stock").collection("Stocks")
+            .where("Sector","==",Sector)
+            .where("SectorPos","<=", cutoff)
+            .orderBy("SectorPos").limit(NumOfSymbols).get()
+        Sectors.forEach(function(doc){
+            symbols.push(doc.data().Symbol)
+        })
+
+    }
+    else{
+
+        let StockInfo =  await db.collection("Ticker-Info").doc("Stock").get();
+        let numOfStocks = StockInfo.data().NumOfStocks - 1;
+        let cutoff = Math.floor((Math.random()  * (numOfStocks - NumOfSymbols))+NumOfSymbols);
+        let Stocks = await db.collection("Ticker-Info").doc("Stock").collection("Stocks")
+            .where("RandomPos",">=", cutoff)
+            .orderBy("RandomPos").limit(NumOfSymbols).get()
+        Stocks.forEach(function(Stock){
+            symbols.push(Stock.data().Symbol)
+        })
+
+    }
+    return symbols
 }
 
-// returns current price for all symbols being tracked
-export const getPrices = async (roomID, dayIndex) => {
-  const roomData = await getRoomData(roomID);
-  const symbolNamesArray = roomData.symbols;
+export const makeInvestment = async (roomID, userID, dayIndex, changeArray) => {
+    const prices = await getPrices(roomID, dayIndex);
 
-  var prices = [];
+    if (!(await verifyOk(roomID,userID,dayIndex,changeArray,prices))) return false;
 
-  // save all current prices in an array
-  for (var index = 0; index < symbolNamesArray.length; index++) {
-      const currentName = symbolNamesArray[index];
-      const symbolPricesDoc = await db.collection('Rooms').doc(roomID).collection(currentName).doc('Prices').get();
-      const symbolPricesData = symbolPricesDoc.data();
-      const currentPrice = symbolPricesData.prices[dayIndex];
-      prices.push(currentPrice);
-  }
+    changeCash(roomID,userID,dayIndex,changeArray,prices);
+    changeShares(roomID,userID,dayIndex,changeArray);
 
-  return prices;
+    return true;
 }
 
-export const getDayIndex = async (roomID) => {
-  const roomData = await getRoomData(roomID);
-  return roomData.day_index;
+function randomDate(start, end) {
+    var date = new Date(+start + Math.random() * (end - start));
+    return date;
 }
 
+export const setUpRoom = (NumOfSymbols,Rounds,userID,password,startingMoney = 10000) => {
```

Comment:
Yep it's outdated the version on master doesn't use initsymbols in it, that's called outside. Reason that's done is because for the config page we needed to have them separate so we could check initsymbols response.